### ISSUE TITLE
S2327 FP: cannot combine try-catch when ignoring exception

### ISSUE THREAD

**Issue by Bouke** [SonarSource/sonar-dotnet#3498]
### Description

[S2327 "try" statements with identical "catch" and/or "finally" blocks should be merged](https://rules.sonarsource.com/csharp/RSPEC-2327) shouldn't be raised if statements in both try clauses should be executed, regardless of an exception being thrown.

### Example 1
```c#
try
{
  DoTheFirstThing();
}
catch (InvalidOperationException ex)
{
  // ignore
}

try
{
  DoTheSecondThing();
}
catch (InvalidOperationException ex)
{
  // ignore
}
```

### Example 2
```c#
try
{
  DoTheFirstThing();
}
catch (InvalidOperationException ex)
{
  HandleException(ex);
}

try
{
  DoTheSecondThing();
}
catch (InvalidOperationException ex)
{
  HandleException(ex);
}

void HandleException(Exception ex) => Console.WriteLine(ex.Message);
```

### Expected behaviour

No S2327 should be raised for this code.

In both examples merging the try-catch clauses would change the behaviour of the code. In the code from the repro `DoTheSecondThing()` would be executed regardless of `DoTheFirstThing()` throwing. However after merging, if `DoTheFirstThing()` throws, `DoTheSecondThing()` would **not** be executed.

### Actual behavior

S2327 is raised.

### Known workarounds

Make the catch-clause different. E.g. change the comment.

### Related information

SonarLint for Visual Studio 2019 4.24.0.20196

**Comment by pavel-mikula-sonarsource** [comment]
Hi @Bouke,

Thank you for reporting this case. This questionable rule implies change of the code behavior in it's description. Unfortunately there's nothing we can fix or improve in the implementation of the rule. Except deleting it.

I suggest you deactivate this rule in case it doesn't fit the needs of your project.
